### ISSUE TITLE
[REGRESSION]: PublishCodeCoverageResults V2 doesn't publish with network errors

### ISSUE THREAD

**Issue by gt97** [microsoft/azure-pipelines-tasks#19811]
### New issue checklist

- [X] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [X] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [ ] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

PublishCodeCoverageResults

### Breaking task version

2.238.1

### Last working task version

1.238.3

### Regression Description

Using the V2 version of PublishCodeCoverageResults I have network errors.
I am using a self hosted agent behind a corporate proxy.

No Issue using the V1 version

Looks to me the proxy configuration of the agent is not used.

### Environment type (Please select at least one enviroment where you face this issue)

- [X] Self-Hosted
- [ ] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

windows server 2022

### Relevant log output

```shell
Could not connect to Azure Pipelines. System.Net.Http.HttpRequestException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (dev.azure.com:443)
 ---> System.Net.Sockets.SocketException (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|281_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Services.Common.VssHttpRetryMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync(HttpRequestMessage message, HttpCompletionOption completionOption, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync[T](HttpRequestMessage message, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Location.Client.LocationHttpClient.GetConnectionDataAsync(ConnectOptions connectOptions, Int64 lastChangeId, CancellationToken cancellationToken, Object userState)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetConnectionDataAsync(ConnectOptions connectOptions, Int32 lastChangeId, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.ConnectAsync(ConnectOptions connectOptions, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.EnsureConnectedAsync(ConnectOptions optionsNeeded, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.CheckForServerUpdatesAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.FindServiceDefinitionsAsync(String serviceType, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetResourceLocationsAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientInstanceAsync(Type managedType, Guid serviceIdentifier, CancellationToken cancellationToken, VssHttpRequestSettings settings, DelegatingHandler[] handlers)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientServiceImplAsync(Type requestedType, Guid serviceIdentifier, Func`4 getInstanceAsync, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientAsync[T](CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.TaskExtensions.SyncResult[T](Task`1 task)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClient[T]()
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.ClientFactory.GetClient[T]() in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\ClientFactory.cs:line 20
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.PipelinesTelemetry..ctor(IClientFactory clientFactory, Boolean enableTelemetryCollection) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\PipelinesTelemetry.cs:line 23
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.AzurePipelinesPublisher..ctor(Boolean enableTelemetry) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\AzurePipelinesPublisher.cs:line 58
   at Microsoft.Azure.Pipelines.CoveragePublisher.Program.ProcessCoverage(PublisherConfiguration config, CancellationToken cancellationToken) in D:\a\1\s\src\CoveragePublisher.C
```

### Full task logs with system.debug enabled

Could not connect to Azure Pipelines. System.Net.Http.HttpRequestException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (dev.azure.com:443)
 ---> System.Net.Sockets.SocketException (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|281_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Services.Common.VssHttpRetryMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync(HttpRequestMessage message, HttpCompletionOption completionOption, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync[T](HttpRequestMessage message, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Location.Client.LocationHttpClient.GetConnectionDataAsync(ConnectOptions connectOptions, Int64 lastChangeId, CancellationToken cancellationToken, Object userState)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetConnectionDataAsync(ConnectOptions connectOptions, Int32 lastChangeId, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.ConnectAsync(ConnectOptions connectOptions, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.EnsureConnectedAsync(ConnectOptions optionsNeeded, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.CheckForServerUpdatesAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.FindServiceDefinitionsAsync(String serviceType, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetResourceLocationsAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientInstanceAsync(Type managedType, Guid serviceIdentifier, CancellationToken cancellationToken, VssHttpRequestSettings settings, DelegatingHandler[] handlers)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientServiceImplAsync(Type requestedType, Guid serviceIdentifier, Func`4 getInstanceAsync, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientAsync[T](CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.TaskExtensions.SyncResult[T](Task`1 task)
   at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClient[T]()
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.ClientFactory.GetClient[T]() in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\ClientFactory.cs:line 20
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.PipelinesTelemetry..ctor(IClientFactory clientFactory, Boolean enableTelemetryCollection) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\PipelinesTelemetry.cs:line 23
   at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.AzurePipelinesPublisher..ctor(Boolean enableTelemetry) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\AzurePipelinesPublisher.cs:line 58
   at Microsoft.Azure.Pipelines.CoveragePublisher.Program.ProcessCoverage(PublisherConfiguration config, CancellationToken cancellationToken) in D:\a\1\s\src\CoveragePublisher.C

### Repro steps

```yml
# V2 version (fails)
          - task: PublishCodeCoverageResults@2
            displayName: "Publish code coverage results"
            inputs:
              summaryFileLocation: $(System.DefaultWorkingDirectory)/target/site/jacoco/jacoco.xml
              pathToSources: $(System.DefaultWorkingDirectory)/src/main/java/

# V1 version (works)
          - task: PublishCodeCoverageResults@1
            displayName: "Publish code coverage results"
            inputs:
              summaryFileLocation: $(System.DefaultWorkingDirectory)/target/site/jacoco/jacoco.xml
              pathToSources: $(System.DefaultWorkingDirectory)/src/main/java/
```

**Comment by dengyakui** [comment]
I had the same issue for AzureDevOpsServer. if I switch back to V1 version , it works well.
i guess it was the client SDK's internal constraint for  `VSS_ALLOW_UNSAFE_BASICAUTH`, is there some workaround ? I dont need HTTS for my intranet AzureDevopsServer.

## V2 version
```
2024-04-24T09:30:29.3712808Z ##[debug]Evaluating condition for step: 'Publish code coverage results'
2024-04-24T09:30:29.3714259Z ##[debug]Evaluating: succeeded()
2024-04-24T09:30:29.3714647Z ##[debug]Evaluating succeeded:
2024-04-24T09:30:29.3715495Z ##[debug]=> True
2024-04-24T09:30:29.3715919Z ##[debug]Result: True
2024-04-24T09:30:29.3716315Z ##[section]Starting: Publish code coverage results
2024-04-24T09:30:29.3722810Z ==============================================================================
2024-04-24T09:30:29.3722995Z Task         : Publish code coverage results v2
2024-04-24T09:30:29.3723125Z Description  : Publish any of the code coverage results from a build
2024-04-24T09:30:29.3723299Z Version      : 2.225.0
2024-04-24T09:30:29.3723392Z Author       : Microsoft Corporation
2024-04-24T09:30:29.3723498Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results
2024-04-24T09:30:29.3723687Z ==============================================================================
2024-04-24T09:30:29.4313069Z ##[debug]Using node path: D:\myagent\agent01\externals\node16\bin\node.exe
2024-04-24T09:30:29.5412192Z ##[debug]agent.TempDirectory=D:\myagent\agent01\_work\_temp
2024-04-24T09:30:29.5424931Z ##[debug]loading inputs and endpoints
2024-04-24T09:30:29.5431974Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2024-04-24T09:30:29.5444432Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2024-04-24T09:30:29.5447573Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2024-04-24T09:30:29.5450256Z ##[debug]loading INPUT_FAILIFCOVERAGEEMPTY
2024-04-24T09:30:29.5452666Z ##[debug]loading INPUT_SUMMARYFILELOCATION
2024-04-24T09:30:29.5461392Z ##[debug]loaded 5
2024-04-24T09:30:29.5467403Z ##[debug]Agent.ProxyUrl=undefined
2024-04-24T09:30:29.5468284Z ##[debug]Agent.CAInfo=undefined
2024-04-24T09:30:29.5468842Z ##[debug]Agent.ClientCert=undefined
2024-04-24T09:30:29.5469387Z ##[debug]Agent.SkipCertValidation=undefined
2024-04-24T09:30:29.5483463Z ##[debug]check path : D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\task.json
2024-04-24T09:30:29.5487699Z ##[debug]adding resource file: D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\task.json
2024-04-24T09:30:29.5488755Z ##[debug]system.culture=en-US
2024-04-24T09:30:29.5504700Z ##[debug]summaryFileLocation=my-app/coverage/cobertura-coverage.xml
2024-04-24T09:30:29.5506774Z ##[debug]failIfCoverageEmpty=false
2024-04-24T09:30:29.5507420Z ##[debug]System.DefaultWorkingDirectory=D:\myagent\agent01\_work\7\s
2024-04-24T09:30:29.5507902Z ##[debug]pathToSources=undefined
2024-04-24T09:30:29.5511532Z ##[debug]defaultRoot: 'D:\myagent\agent01\_work\7\s'
2024-04-24T09:30:29.5512203Z ##[debug]findOptions.allowBrokenSymbolicLinks: 'false'
2024-04-24T09:30:29.5512886Z ##[debug]findOptions.followSpecifiedSymbolicLink: 'false'
2024-04-24T09:30:29.5513631Z ##[debug]findOptions.followSymbolicLinks: 'false'
2024-04-24T09:30:29.5514210Z ##[debug]findOptions.skipMissingFiles: 'undefined'
2024-04-24T09:30:29.5515094Z ##[debug]matchOptions.debug: 'false'
2024-04-24T09:30:29.5515797Z ##[debug]matchOptions.nobrace: 'true'
2024-04-24T09:30:29.5516299Z ##[debug]matchOptions.noglobstar: 'false'
2024-04-24T09:30:29.5516831Z ##[debug]matchOptions.dot: 'true'
2024-04-24T09:30:29.5517333Z ##[debug]matchOptions.noext: 'false'
2024-04-24T09:30:29.5517752Z ##[debug]matchOptions.nocase: 'true'
2024-04-24T09:30:29.5518280Z ##[debug]matchOptions.nonull: 'false'
2024-04-24T09:30:29.5518909Z ##[debug]matchOptions.matchBase: 'false'
2024-04-24T09:30:29.5519331Z ##[debug]matchOptions.nocomment: 'false'
2024-04-24T09:30:29.5519749Z ##[debug]matchOptions.nonegate: 'false'
2024-04-24T09:30:29.5520171Z ##[debug]matchOptions.flipNegate: 'false'
2024-04-24T09:30:29.5520614Z ##[debug]pattern: 'my-app/coverage/cobertura-coverage.xml'
2024-04-24T09:30:29.5536184Z ##[debug]findPath: 'D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml'
2024-04-24T09:30:29.5538014Z ##[debug]statOnly: 'true'
2024-04-24T09:30:29.5538401Z ##[debug]found 1 paths
2024-04-24T09:30:29.5538797Z ##[debug]applying include pattern
2024-04-24T09:30:29.5539290Z ##[debug]adjustedPattern: 'D:\myagent\agent01\_work\7\s\my-app/coverage/cobertura-coverage.xml'
2024-04-24T09:30:29.5544602Z ##[debug]1 matches
2024-04-24T09:30:29.5545131Z ##[debug]1 final results
2024-04-24T09:30:29.5561538Z Found 1 result(s) matching pattern: my-app/coverage/cobertura-coverage.xml
2024-04-24T09:30:29.5565904Z D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml
2024-04-24T09:30:29.5568443Z ##[debug]Agent.Version=2.217.2
2024-04-24T09:30:29.5569061Z ##[debug]Agent.TempDirectory=D:\myagent\agent01\_work\_temp
2024-04-24T09:30:29.5576377Z ##[debug]which 'dotnet'
2024-04-24T09:30:29.5730663Z ##[debug]found: 'C:\Program Files\dotnet\dotnet.exe'
2024-04-24T09:30:29.5732747Z ##[debug]which 'D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe'
2024-04-24T09:30:29.5735051Z ##[debug]found: 'D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe'
2024-04-24T09:30:29.5736227Z ##[debug]D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe arg: D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml
2024-04-24T09:30:29.5737677Z ##[debug]D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe arg: --reportDirectory
2024-04-24T09:30:29.5738682Z ##[debug]D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe arg: D:\myagent\agent01\_work\_temp\5ee5a6bf-c392-47e2-8869-7f6644790722
2024-04-24T09:30:29.5743968Z ##[debug]SystemVssConnection auth param AccessToken = ***
2024-04-24T09:30:29.5744584Z ##[debug]System.TeamFoundationCollectionUri=http://quintos:8080/tfs/DefaultCollection/
2024-04-24T09:30:29.5745368Z ##[debug]Build.BuildId=137
2024-04-24T09:30:29.5745995Z ##[debug]Build.ContainerId=20967
2024-04-24T09:30:29.5746715Z ##[debug]Agent.TempPath=undefined
2024-04-24T09:30:29.5747489Z ##[debug]System.TeamProjectId=808aa72b-83b4-4261-afc9-146fe01e6a46
2024-04-24T09:30:29.5748074Z ##[debug]PIPELINES_COVERAGEPUBLISHER_DEBUG=undefined
2024-04-24T09:30:29.5748883Z ##[debug]exec tool: D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe
2024-04-24T09:30:29.5749602Z ##[debug]arguments:
2024-04-24T09:30:29.5750162Z ##[debug]   D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml
2024-04-24T09:30:29.5750715Z ##[debug]   --reportDirectory
2024-04-24T09:30:29.5751169Z ##[debug]   D:\myagent\agent01\_work\_temp\5ee5a6bf-c392-47e2-8869-7f6644790722
2024-04-24T09:30:29.5751960Z [command]"D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe" D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml --reportDirectory D:\myagent\agent01\_work\_temp\5ee5a6bf-c392-47e2-8869-7f6644790722
2024-04-24T09:30:29.8272396Z Could not connect to Azure Pipelines. System.InvalidOperationException: Basic authentication requires a secure connection to the server.
2024-04-24T09:30:29.8273614Z    at Microsoft.VisualStudio.Services.Common.VssBasicCredential.OnCreateTokenProvider(Uri serverUrl, IHttpResponse response)
2024-04-24T09:30:29.8274542Z    at Microsoft.VisualStudio.Services.Common.IssuedTokenCredential.CreateTokenProvider(Uri serverUrl, IHttpResponse response, IssuedToken failedToken)
2024-04-24T09:30:29.8276310Z    at Microsoft.VisualStudio.Services.Common.VssCredentials.TryGetTokenProvider(Uri serverUrl, IssuedTokenProvider& provider)
2024-04-24T09:30:29.8277150Z    at Microsoft.VisualStudio.Services.Common.VssHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
2024-04-24T09:30:29.8277718Z    at Microsoft.VisualStudio.Services.Common.VssHttpRetryMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
2024-04-24T09:30:29.8278660Z    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
2024-04-24T09:30:29.8279486Z    at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync(HttpRequestMessage message, HttpCompletionOption completionOption, Object userState, CancellationToken cancellationToken)
2024-04-24T09:30:29.8280155Z    at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync[T](HttpRequestMessage message, Object userState, CancellationToken cancellationToken)
2024-04-24T09:30:29.8280971Z    at Microsoft.VisualStudio.Services.Location.Client.LocationHttpClient.GetConnectionDataAsync(ConnectOptions connectOptions, Int64 lastChangeId, CancellationToken cancellationToken, Object userState)
2024-04-24T09:30:29.8281688Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetConnectionDataAsync(ConnectOptions connectOptions, Int32 lastChangeId, CancellationToken cancellationToken)
2024-04-24T09:30:29.8282335Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.ConnectAsync(ConnectOptions connectOptions, CancellationToken cancellationToken)
2024-04-24T09:30:29.8283045Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.EnsureConnectedAsync(ConnectOptions optionsNeeded, CancellationToken cancellationToken)
2024-04-24T09:30:29.8283739Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.CheckForServerUpdatesAsync(CancellationToken cancellationToken)
2024-04-24T09:30:29.8284320Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.FindServiceDefinitionsAsync(String serviceType, CancellationToken cancellationToken)
2024-04-24T09:30:29.8284900Z    at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetResourceLocationsAsync(CancellationToken cancellationToken)
2024-04-24T09:30:29.8285573Z    at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientInstanceAsync(Type managedType, Guid serviceIdentifier, CancellationToken cancellationToken, VssHttpRequestSettings settings, DelegatingHandler[] handlers)
2024-04-24T09:30:29.8286423Z    at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientServiceImplAsync(Type requestedType, Guid serviceIdentifier, Func`4 getInstanceAsync, CancellationToken cancellationToken)
2024-04-24T09:30:29.8287420Z    at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClientAsync[T](CancellationToken cancellationToken)
2024-04-24T09:30:29.8287853Z    at Microsoft.VisualStudio.Services.WebApi.TaskExtensions.SyncResult[T](Task`1 task)
2024-04-24T09:30:29.8288222Z    at Microsoft.VisualStudio.Services.WebApi.VssConnection.GetClient[T]()
2024-04-24T09:30:29.8288779Z    at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.ClientFactory.GetClient[T]() in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\ClientFactory.cs:line 20
2024-04-24T09:30:29.8289773Z    at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.PipelinesTelemetry..ctor(IClientFactory clientFactory, Boolean enableTelemetryCollection) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\PipelinesTelemetry.cs:line 23
2024-04-24T09:30:29.8290677Z    at Microsoft.Azure.Pipelines.CoveragePublisher.Publishers.DefaultPublisher.AzurePipelinesPublisher..ctor(Boolean enableTelemetry) in D:\a\1\s\src\CoveragePublisher\Publishers\DefaultPublisher\AzurePipelinesPublisher.cs:line 58
2024-04-24T09:30:29.8291573Z    at Microsoft.Azure.Pipelines.CoveragePublisher.Program.ProcessCoverage(PublisherConfiguration config, CancellationToken cancellationToken) in D:\a\1\s\src\CoveragePublisher.Console\Program.cs:line 43
2024-04-24T09:30:29.8292568Z ##[debug]task result: Failed
2024-04-24T09:30:29.8300723Z ##[debug]Processed: ##vso[task.complete result=Failed;]
2024-04-24T09:30:29.8360662Z ##[debug]Exit code 0 received from tool 'D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe'
2024-04-24T09:30:29.8366534Z ##[debug]STDIO streams have closed for tool 'D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\2.225.0\node_modules\coveragepublisher\CoveragePublisher\CoveragePublisher.Console.exe'
2024-04-24T09:30:29.8445362Z ##[section]Finishing: Publish code coverage results

```


### v1 version
```
2024-04-24T09:19:56.2984327Z ##[section]Starting: Publish code coverage results
2024-04-24T09:19:56.2991177Z ==============================================================================
2024-04-24T09:19:56.2991371Z Task         : Publish code coverage results
2024-04-24T09:19:56.2991487Z Description  : Publish Cobertura or JaCoCo code coverage results from a build
2024-04-24T09:19:56.2991640Z Version      : 1.219.0
2024-04-24T09:19:56.2991743Z Author       : Microsoft Corporation
2024-04-24T09:19:56.2991859Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results
2024-04-24T09:19:56.2992048Z ==============================================================================
2024-04-24T09:19:56.8180730Z [command]"C:\Program Files\dotnet\dotnet.exe" D:\myagent\agent01\_work\_tasks\PublishCodeCoverageResults_2a7ebc54-c13e-490e-81a5-d7561ab7cd97\1.219.0\netcoreapp2.0\ReportGenerator.dll -reports:my-app/coverage/cobertura-coverage.xml -targetdir:D:\myagent\agent01\_work\_temp\cchtml -reporttypes:HtmlInline_AzurePipelines
2024-04-24T09:19:56.8766113Z 2024-04-24T17:19:56: Arguments
2024-04-24T09:19:56.8766640Z 2024-04-24T17:19:56:  -reports:my-app/coverage/cobertura-coverage.xml
2024-04-24T09:19:56.8768579Z 2024-04-24T17:19:56:  -targetdir:D:\myagent\agent01\_work\_temp\cchtml
2024-04-24T09:19:56.8769159Z 2024-04-24T17:19:56:  -reporttypes:HtmlInline_AzurePipelines
2024-04-24T09:19:57.1396094Z 2024-04-24T17:19:57: Writing report file 'D:\myagent\agent01\_work\_temp\cchtml\index.html'
2024-04-24T09:19:57.1765134Z 2024-04-24T17:19:57: Report generation took 0.2 seconds
2024-04-24T09:19:57.1854207Z Generated code coverage html report: D:\myagent\agent01\_work\_temp\cchtml
2024-04-24T09:19:57.1882437Z Reading code coverage summary from 'D:\myagent\agent01\_work\7\s\my-app\coverage\cobertura-coverage.xml'
2024-04-24T09:19:57.2523554Z ##[section]Async Command Start: Publish code coverage
2024-04-24T09:19:57.2523795Z Publishing coverage summary data to TFS server.
2024-04-24T09:19:57.2524206Z  Lines- 1 of 12 covered.
2024-04-24T09:19:57.2524433Z  Branches- 0 of 4 covered.
2024-04-24T09:19:57.5865696Z Modifying Cobertura Index file
2024-04-24T09:19:57.5865907Z Publishing code coverage files to TFS server.
2024-04-24T09:19:57.5866058Z Uploading 6 files
2024-04-24T09:19:57.5866201Z File upload succeed.
2024-04-24T09:19:57.5866420Z Published 'D:\myagent\agent01\_work\_temp\cchtml' as artifact 'Code Coverage Report_135'
2024-04-24T09:19:57.5866854Z ##[section]Async Command End: Publish code coverage
2024-04-24T09:19:57.5868210Z ##[section]Finishing: Publish code coverage results
```

**Comment by gt97** [comment]
I am using Azure Devops cloud so all connection are https

Bye

**Comment by KaiBNET** [comment]
Same problem here with Azure Devops cloud.

**Comment by mkot02** [comment]
Still broken in 2.240.1. We're running self-hosted agent behind corporate proxy with cloud-hosted Azure DevOps. Going back to @1 "solved" the issue.

**Comment by trysetnull** [comment]
Still broken in Version `2.242.1`. Same as above we're running a self-hosted agent behind a corporate web-proxy with a cloud-hosted Azure DevOps instance. Using `@1` (Version `1.244.1`) works but generates the following warning:

```
Error analysis
Publish Code Coverage to ADOS (PublishCodeCoverageResults)

New V2 version of task publishing code coverage results is available to all our customers now. We highly recommend to stop using the V1 version and migrate to V2 version (https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/publish-code-coverage-results-v2). For more details, see - https://devblogs.microsoft.com/devops/new-pccr-task.
```

Can someone look into this?!

**Comment by JacketsMask** [comment]
We're experiencing the same issue with version 2.242.1.

**Comment by ichekoti** [comment]
Exact same issue, self hosted agents behind corp proxy. 

Starting: Publish code coverage from  JACOCO Reports
==============================================================================
Task         : Publish code coverage results v2
Description  : Publish any of the code coverage results from a build
Version      : 2.242.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-code-coverage-results

**Comment by ACOR03** [comment]
Still the same issue with version 2.248.1
Configuration : self hosted agent behind a corporate proxy.

**Comment by sambit-diff** [comment]
Hi, any updates on this please! we are getting the same issue using task v2.252.1 with self-hosted agents behind proxy.

**Comment by kbmuellerdo** [comment]
Hey,
same in our environment. We can confirm that proxy settings will be still ignored and connections went direct which are blocked by our fw. Is there any ETA for a possible fix?

**Comment by patpatx** [comment]
Tried investigating this issue last year, the problem I noticed is that the proxy configuration is not being passed down to the actual task.

ive commited a possible fix last year to pass down the HTTP_PROXY and HTTPS_PROXY config but still pending for PR

https://github.com/microsoft/azure-pipelines-tasks/pull/19826

**Comment by Gabboxl** [comment]
Still experiencing this issue with self-hosted agents behind a proxy. Any updates on this, please?

**Comment by gt97** [comment]
Issue still present. When will it be solved?

**Comment by kbmuellerdo** [comment]
Same on our side. Is there any ETA?

**Comment by T-Ernst** [comment]
For us, the issue has been resolved.

We are running self-hosted agents behind a corporate web-proxy with a cloud-hosted Azure DevOps instance.

Using version `2.259.0` of the `PublishCodeCoverageResults@2` now successfully uploads the code coverage report.
Note: we have set the environment variables `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY`.

According to #21011 from version `2.257.0` this issue should be fixed.

**Comment by gt97** [comment]
I made some new tests according to @T-Ernst suggestion. With this Work Around works (it didn't get the automatic proxy from the agent)
